src/index.js

```diff
@@ -35,8 +52,8 @@ function update(state=initialState, action) {
 
 // Syncing
 
-function locationToString(location) {
-  return location.pathname + location.search + location.hash;
+function locationsAreEqual(a, b) {
+  return a.path === b.path && deepEqual(a.state, b.state);
```

Comment:
It's basically a copy of https://github.com/rackt/history/blob/master/modules/createHistory.js#L12-L18